### DISCUSSION TITLE
Update Bootloader in eMMC Boot Partitions and without erasing the complete boot0/1 partition

### DISCUSSION THREAD

**Discussion by bigstefan** [rauc/rauc#1580] [Q&A]
Hi
I'm doing a prototype for RAUC updates for our product. We are using eMMC and like to update U-Boot.
In the boot0 partition there are additional information stored beside spl, u-boot and u-boot-env.
Is there a generic way to only erase the first section of the boot0 or boot1 and keep the second section untouched?

In the doc and the code I could not find a possibility for `type=boot-emmc`. 
With `type=boot-mbr-switch`here are `region-start` and `region-size`.
So I hacked the `update_handler.c clear_slot()` and erased only the first half of the partition, this worked.

Do you see a better way of doing this?

- There is an other possibility with pre/post hook to save and restore the data, but this is not fail-safe.
- An other possibility would be to store the additional data not only in boot0 but also in boot1. Here I have the problem to change all accessor of the data to read and write from the active boot-partiton. This is possible but a challenging job in third party code.

Thanks for your help
Best Regards
Stefan

**Comment by ejoerns** [comment]
> An other possibility would be to store the additional data not only in boot0 but also in boot1. Here I have the problem to change all accessor of the data to read and write from the active boot-partiton. This is possible but a challenging job in third party code.

If the data is not meant to be redundant, I would not store them in a boot partition at all (if possible).

An alternative would be to generate an image containing the bootloader and all the other data that should be part of this partition.

A very similar requirement / discussion for what you tried, can be found in https://github.com/rauc/rauc/issues/780

**Reply by bigstefan** [reply]
Hi 
Thanks for the answer and sorry that I did not see #780
Yes it is the same issue we also have modules from Toradex with production data. In addition we added our production data next to them. Changing this will be very difficult and we loose the backwards compatibility what I doubt will be accepted.

The data production data it is device specific so we cannot put it in the image.

Adding a parameter for `size-limit` would this be still a valid option?

**Reply by ejoerns** [reply]
> Adding a parameter for size-limit would this be still a valid option?

Since this would not change any existing behavior when unset, while enabling support for these specific use-cases, I'd say yes.

**Reply by jluebbe** [reply]
You can add general purpose partitions on eMMCs if you need to store data that is neither a bootloader nor part of the main firmware image.
